radix-engine-common/src/math/bnum_integer.rs

```diff
@@ -59,6 +59,41 @@ macro_rules! types {
                     pub const BITS: u32 = $wrap::BITS as u32;
                     pub const BYTES: u32 = $wrap::BYTES as u32;
                     pub const N: usize = ($wrap::BYTES / 8) as usize;
+
+                    pub const fn const_from_str(from: &str) -> Result<Self, [<Parse $t Error>] > {
+                        match $wrap::from_str_radix(from, 10) {
+                            Ok(val) => Ok(Self(val)),
+                            Err(_) => Err([<Parse $t Error>]::InvalidDigit),
+                        }
+                    }
+                    pub const fn parse_str(from: &str) -> Self {
+                        match Self::const_from_str(from) {
+                            Ok(val) => val,
+                            Err(_) => panic!("Invalid string given"),
+                        }
+                    }
+
+                    pub const fn const_mul(self, other: $t) -> Self {
+                        match self.0.checked_mul(other.0) {
+                            Some(val) => Self(val),
+                            None => panic!("Overflow"),
+                        }
+                    }
+
+                    pub const fn const_div(self, other: $t) -> Self {
+                        match self.0.checked_div(other.0) {
+                            Some(val) => Self(val),
+                            None => panic!("Overflow"),
+                        }
+                    }
+
+
+                    pub const fn const_pow(self, pow: u32) -> Self {
+                        match self.0.checked_pow(pow) {
+                            Some(val) => Self(val),
+                            None => panic!("Overflow"),
```

Comment:
Suggest we do not add all these overflow variants, at least not close to release.

Reply:
Refactored whole implementation. Hope it looks better :)